packages/ui/src/components/avatar/avatar.web.tsx

```diff
@@ -1,114 +1,155 @@
-import { forwardRef } from 'react';
+import {
+  ComponentPropsWithoutRef,
+  ElementRef,
+  ReactElement,
+  cloneElement,
+  forwardRef,
+} from 'react';
 
 import * as RadixAvatar from '@radix-ui/react-avatar';
-import { type RecipeVariantProps, css, cva } from 'leather-styles/css';
-import { type HTMLStyledProps, styled } from 'leather-styles/jsx';
+import { styled } from 'leather-styles/jsx';
 
-export const defaultFallbackDelay = 600;
-export function getAvatarFallback(val: string) {
-  return val.slice(0, 2);
+import { isDefined } from '@leather.io/utils';
+
+import { IconProps } from '../../icons/icon/create-icon.web';
+import { AvatarSize, AvatarVariant, defaultFallbackDelay, iconSizeMap } from './avatar.shared';
+
+type AvatarElement = ElementRef<typeof AvatarRoot>;
+
+export interface AvatarProps extends ComponentPropsWithoutRef<typeof AvatarRoot> {
```

Comment:
Can we move this into a shared `types` and share more between web and native?

Reply:
I'm afraid not–this is a web/panda specific type signature.